supervisor/core.py

```diff
@@ -182,8 +182,8 @@ async def start(self):
         # Mark booted partition as healthy
         await self.sys_os.mark_healthy()
 
-        # On release channel, try update itself
-        if self.sys_supervisor.need_update:
+        # On release channel, try update itself if auto update enabled
+        if self.sys_supervisor.need_update and self.sys_updater.auto_update:
```

Comment:
We need to change how `need_update` works with this PR.

This is the current implementation:

```
    @property
    def need_update(self) -> bool:
        """Return True if an update is available."""
        if self.sys_dev:
            return False

        try:
            return self.version < self.latest_version
        except (AwesomeVersionException, TypeError):
            return False
```

This needs to return True by default until we know we got a good response for an updated version file

Reply:
Hm yea good point. Will that have an affect on the front end though? It shows update notifications anytime that is true right? I see your point though, this check needs to assume there is an update until it's sure there isn't. I'll adjust to do that.

Reply:
If there is an issue in the frontend I’ll fix that

Reply:
We have the update endpoints for the core. We properly only need adjust this API endpoint to see only supervisor and not others, so they can update supervisor and after that it start to show the other updates?

Reply:
For newer installations it should not use that attribute at all, this should only be an issue for older core installations before that endpoint and before update entities

Reply:
Well its not just frontend though. LIke if I make `need_update` return true until we have checked for an update think about what happens for the majority of users that have `auto_update` enabled in this spot. It will actually try and execute an update there. We don't want that either.

I think we have two options here:
1) This method becomes `Optional[bool]` and return `None` when we don't know if there's an update because the version file hasn't been pulled.
2) Leave this method alone and instead the `SUPERVISOR_UPDATED` job condition prevents calls to the method when there is a supervisor update or it sees we haven't pulled the version file yet.

Reply:
After a bit of discussion what we went with is a new evaluation for `supervisor_version`. We only support the latest supervisor version so if you are running on an old version of supervisor and have auto-update disabled then the evaluation will mark the system as unsupported. 

No that doesn't stop someone who blocks the version file and prevents supervisor from knowing about updates. But if a user doing that tries to submit an issue the first thing we'll see is that the supervisor version is out of date. So we'll know its an unsupported system even if the check has been worked around.

This also deals with hacks like ignoring the `SUPERVISOR_UPDATED` job condition then un-ignoring it after an update. The system will be marked unsupported regardless because supervisor is out of date.